source/payload-interfaces/index.rst

```diff
@@ -86,167 +109,207 @@ type EFI_RESOURCE_MEMORY_RESERVED.
 
-Point to the ACPI RSDP table. The ACPI table need follow ACPI specification verson 2.0 or above.
+SMBIOS Table
+^^^^^^^^^^^^
 
-SMBIOS Table HOB
-~~~~~~~~~~~~~~~~
+The bootloader might pass SMBIOS table to the payload. So that the payload could get the platform information from the table.
 
-The bootloader might pass SMBIOS table through the GUID HOB to the
-payload. So that the payload could get the platform information from
-the table.
+**GUID**
 
-**HOB GUID**::
+::
 
-  #define    SMBIOS_TABLE_GUID    {0xeb9d2d31, 0x2d88, 0x11d3, {0x9a, 0x16, 0x0, 0x90, 0x27, 0x3f, 0xc1, 0x4d}}
+  gPldSmbios3TableGuid = { 0x92b7896c, 0x3362, 0x46ce, { 0x99, 0xb3, 0x4f, 0x5e, 0x3c, 0x34, 0xeb, 0x42 } }
 
-  #define    SMBIOS3_TABLE_GUID   {0xf2fd1544, 0x9794, 0x4a2c, {0x99, 0x2e, 0xe5, 0xbb, 0xcf, 0x20, 0xe3, 0x94}}
+  gPldSmbiosTableGuid = { 0x590a0d26, 0x06e5, 0x4d20, { 0x8a, 0x82, 0x59, 0xea, 0x1b, 0x34, 0x98, 0x2d } }
 
-.. Note:: These GUIDs reuse the same GUIDs defined in UEFI spec chapter 4.6 EFI Configuration Table
+**Structure**
 
-**Hob Interface Structure**::
+::
 
   #pragma pack (1)
 
-  //
-  // Bootloader SMBIOS table hob
-  //
   typedef struct {
-    EFI_HOB_GUID_TYPE   Header;
-    UINT64              TableAddress;
-  } SMBIOS_TABLE_HOB;
+    PLD_GENERIC_HEADER   PldHeader;
+    EFI_PHYSICAL_ADDRESS SmBiosEntryPoint;
+  } PLD_SMBIOS_TABLE;
 
   #pragma pack()
 
 **Member Description**
 
-``Header``
+``PldHeader``
+
+PldHeader.Revision is 0.
+
+``SmBiosEntryPoint``
+
+Points to the SMBIOS table in SMBIOS 3.0+ format if GUID is ``gPldSmbios3TableGuid``.
 
-Header.Name set to SMBIOS_TABLE_GUID if SMBIOS table from
-TableAddress follows the format defined by SMBIOS_TABLE_ENTRY_POINT,
-or set to SMBIOS3_TABLE_GUID if SMBIOS table from TableAddress
-follows the format defied by SMBIOS_TABLE_3_0_ENTRY_POINT. See
-section 6.5 EFI_HOB_GUID_TYPE.
+Points to the SMBIOS table in SMBIOS 2.x format if GUID is ``gPldSmbiosTableGuid``.
 
-``AcpiTableAddress``
+DEVICE TREE
+^^^^^^^^^^^
 
-Point to the SMBIOS table entry point.
+The bootloader might pass Device Tree to the payload. So that the payload could get the platform information from the table.
 
-DEVICE TREE HOB
-~~~~~~~~~~~~~~~
+**GUID**
 
-The bootloader might pass Device Tree through the GUID HOB to the
-payload. So that the payload could get the platform information from
-the table.
+::
 
-**HOB GUID**::
+  gPldDeviceTreeGuid = {0x6784b889, 0xb13c, 0x4c3b, {0xae, 0x4b, 0xf, 0xa, 0x2e, 0x32, 0xe, 0xa3}}
 
-  #define    DEVICE_TREE_GUID    {0x6784b889, 0xb13c, 0x4c3b, {0xae, 0x4b, 0xf, 0xa, 0x2e, 0x32, 0xe, 0xa3}}
+**Structure**
 
-**Hob Interface Structure**::
+::
 
   #pragma pack (1)
 
-  //
-  // Bootloader Device Tree hob
-  //
   typedef struct {
-    EFI_HOB_GUID_TYPE     Header;
-    UINT64                DeviceTreeAddress;
-  } DEVICE_TREE_HOB;
+    PLD_GENERIC_HEADER   PldHeader;
+    UINT64               DeviceTreeAddress;

```

Comment:
Should we use EFI_PHYSICAL_ADDRESS to be consistent ?

Reply:
yes